azure/resources/pf-key-vault/functions/public/new-pfkeyvaultbuild.ps1

```diff
@@ -1,16 +1,31 @@
 Function New-PfKeyVaultBuild
 {
+    <#
     Param (
         [Parameter(Mandatory = $false)] 
         [PfKeyVaultOptions] $Options
     )   
     # use this to get the parameter set name
     if($Options)
     {
-        [PfKeyVaultBuildState]::New($Options)
+        $kv = [PfKeyVaultBuildState]::New($Options)
     }
     else {
-        [PfKeyVaultBuildState]::New()
+        $kv = [PfKeyVaultBuildState]::New()
     }
+    $global:_PfDeploymentContext.Resources.Add($kv) | Out-Null
+    return $kv
+    #>
+
+    if($global:_PfAzureContext)
+    {
+        $kv = [PfKeyVaultBuildState]::New($global:_PfAzureContext)
+    }
+    else{
+        $kv = [PfKeyVaultBuildState]::New()
+    }
+    $global:_PfDeploymentContext.Resources.Add($kv) | Out-Null
+    return $kv
```

Comment:
No more options object needed to create a build state.  The build state builds the options by default or by the pfazurecontext you pass in.  If you want to change the options you can manage those options off the buildstate object it returns.